src/components/Modal/Modal.jsx

```diff
@@ -0,0 +1,254 @@
+import React, { cloneElement, useCallback, useEffect, useMemo } from "react";
+import ReactDOM from "react-dom";
+import PropTypes from "prop-types";
+import cx from "classnames";
+import styles from "./Modal.module.scss";
+import { useA11yDialog } from "./a11yDialog";
+import { ModalContent, ModalFooter, ModalHeader } from "components";
+import { clearAllBodyScrollLocks, disableBodyScroll, enableBodyScroll } from "body-scroll-lock";
+import useAnimationProps from "components/Modal/useAnimationProps";
+import { useKeyEvent } from "../../hooks";
+
+export const MODAL_WIDTH = {
+  DEFAULT: "default",
+  FULL_WIDTH: "full_width"
+};
+
+const isModalHeader = child => child.type === ModalHeader;
+const isModalContent = child => child.type === ModalContent;
+const isModalFooter = child => child.type === ModalFooter;
+
+const Modal = ({
+  classNames,
+  id,
+  show,
+  title,
+  description,
+  onClose,
+  isAlertDialog,
+  children,
+  triggerElement,
+  width,
+  hideCloseButton,
+  closeButtonAriaLabel
+}) => {
+  const [instance, attr] = useA11yDialog({
+    id,
+    isAlertDialog
+  });
+
+  useKeyEvent({
+    callback: event => {
+      if (instance?.$el.contains(document.activeElement)) {
+        event.stopPropagation();
+        closeDialogIfNeeded();
+      }
+    },
+    capture: true,
+    keys: ["Escape"]
+  });
+
+  const closeDialogIfNeeded = useCallback(() => {
+    if (!isAlertDialog) {
+      onClose();
+    }
+  }, [isAlertDialog, onClose]);
+
+  const getAnimationProps = useAnimationProps(triggerElement, instance);
+
+  // clear all scroll locks on unmount (just to be safe)
+  useEffect(() => {
```

Comment:
it might be a matter of preference, but consider extracting these 3 `useEffect`s to a separate hook (something like `useLockedBodyScroll` or whatever)

Reply:
I agree. I really like to use custom hooks for separate different logic part which are not related or depends on each-other for readability. 
But it's not a blocker ofcourse.

Reply:
done